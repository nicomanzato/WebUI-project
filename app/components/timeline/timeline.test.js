import React from 'react';
import Timeline from './timeline';
import PostMock from './../../store/post/mock/postMock'
import PostMock2 from './../../store/post/mock/postMock2'
import PostMock3 from './../../store/post/mock/postMock3'
import { shallow, mount, ReactWrapper } from 'enzyme';

describe('post screen', () => {

  it('should render correctly when done loading', () => {
    const wrapper = shallow(
      <Timeline
        data={[PostMock, PostMock2, PostMock3]}
        loading={false}
        onComponentMount={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when loading', () => {
    const wrapper = shallow(
      <Timeline
        data={[]}
        loading={true}
        onComponentMount={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
