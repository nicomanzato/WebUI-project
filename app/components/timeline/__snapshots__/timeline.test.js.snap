// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post screen should render correctly when done loading 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <PostList
    data={
      Array [
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Thu Dec 20 14:06:56 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "twitter.com/i/web/status/1…",
                "expanded_url": "https://twitter.com/i/web/status/1075754427368333313",
                "indices": Array [
                  117,
                  140,
                ],
                "url": "https://t.co/WG9oBkfoFF",
              },
            ],
            "user_mentions": Array [],
          },
          "favorite_count": 0,
          "favorited": false,
          "geo": null,
          "id": 1075754427368333300,
          "id_str": "1075754427368333313",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 0,
          "retweeted": false,
          "source": "<a href=\\"http://twitter.com\\" rel=\\"nofollow\\">Twitter Web Client</a>",
          "text": "Por una medida de Trump, Rusia advirtió que \\"hay peligro\\" de una guerra nuclear: \\"Puede morir toda la civilización\\"… https://t.co/WG9oBkfoFF",
          "truncated": true,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Mon Mar 23 12:35:07 +0000 2009",
            "default_profile": true,
            "default_profile_image": false,
            "description": "Noticias las 24 horas, por el canal líder en audiencia y credibilidad de Argentina",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "tn.com.ar",
                    "expanded_url": "http://www.tn.com.ar",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/TX6TF1nOUs",
                  },
                ],
              },
            },
            "favourites_count": 1,
            "follow_request_sent": false,
            "followers_count": 4677787,
            "following": false,
            "friends_count": 124,
            "geo_enabled": false,
            "has_extended_profile": false,
            "id": 25992212,
            "id_str": "25992212",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 9085,
            "location": "Argentina",
            "name": "TN - Todo Noticias",
            "notifications": false,
            "profile_background_color": "666666",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/25992212/1533675118",
            "profile_image_url": "http://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_link_color": "D51D2E",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "314BB9",
            "profile_text_color": "000000",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "todonoticias",
            "statuses_count": 84280,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/TX6TF1nOUs",
            "utc_offset": null,
            "verified": true,
          },
        },
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Fri Dec 21 12:52:02 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "media": Array [
              Object {
                "display_url": "pic.twitter.com/GuK1rpkdeJ",
                "expanded_url": "https://twitter.com/clarincom/status/1076097967147360257/photo/1",
                "id": 1076097964253298700,
                "id_str": "1076097964253298688",
                "indices": Array [
                  100,
                  123,
                ],
                "media_url": "http://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                "media_url_https": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 630,
                    "resize": "fit",
                    "w": 1200,
                  },
                  "medium": Object {
                    "h": 630,
                    "resize": "fit",
                    "w": 1200,
                  },
                  "small": Object {
                    "h": 357,
                    "resize": "fit",
                    "w": 680,
                  },
                  "thumb": Object {
                    "h": 150,
                    "resize": "crop",
                    "w": 150,
                  },
                },
                "type": "photo",
                "url": "https://t.co/GuK1rpkdeJ",
              },
            ],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "clarin.com/espectaculos/m…",
                "expanded_url": "https://www.clarin.com/espectaculos/microteatro-verano-agenda-cargada-pasar-meses-calor_0_Qh15Av7jj.html?utm_term=Autofeed&utm_medium=Social&utm_source=Twitter#Echobox=1545395977",
                "indices": Array [
                  76,
                  99,
                ],
                "url": "https://t.co/3V4dX1OSiJ",
              },
            ],
            "user_mentions": Array [],
          },
          "extended_entities": Object {
            "media": Array [
              Object {
                "display_url": "pic.twitter.com/GuK1rpkdeJ",
                "expanded_url": "https://twitter.com/clarincom/status/1076097967147360257/photo/1",
                "id": 1076097964253298700,
                "id_str": "1076097964253298688",
                "indices": Array [
                  100,
                  123,
                ],
                "media_url": "http://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                "media_url_https": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 630,
                    "resize": "fit",
                    "w": 1200,
                  },
                  "medium": Object {
                    "h": 630,
                    "resize": "fit",
                    "w": 1200,
                  },
                  "small": Object {
                    "h": 357,
                    "resize": "fit",
                    "w": 680,
                  },
                  "thumb": Object {
                    "h": 150,
                    "resize": "crop",
                    "w": 150,
                  },
                },
                "type": "photo",
                "url": "https://t.co/GuK1rpkdeJ",
              },
            ],
          },
          "favorite_count": 0,
          "favorited": false,
          "geo": null,
          "id": 1076097967147360300,
          "id_str": "1076097967147360257",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 0,
          "retweeted": false,
          "source": "<a href=\\"https://www.echobox.com\\" rel=\\"nofollow\\">Echobox Social</a>",
          "text": "Microteatro de verano, una agenda cargada para pasar los meses de más calor https://t.co/3V4dX1OSiJ https://t.co/GuK1rpkdeJ",
          "truncated": false,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Fri Aug 10 15:14:11 +0000 2007",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "clarin.com",
                    "expanded_url": "http://www.clarin.com",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/1RuczZo0FK",
                  },
                ],
              },
            },
            "favourites_count": 391,
            "follow_request_sent": false,
            "followers_count": 2810398,
            "following": true,
            "friends_count": 265,
            "geo_enabled": true,
            "has_extended_profile": false,
            "id": 8105922,
            "id_str": "8105922",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 10835,
            "location": "Buenos Aires, Argentina",
            "name": "Clarín",
            "notifications": false,
            "profile_background_color": "131516",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1545358874",
            "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
            "profile_link_color": "FF0324",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "EFEFEF",
            "profile_text_color": "333333",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "clarincom",
            "statuses_count": 301824,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/1RuczZo0FK",
            "utc_offset": null,
            "verified": false,
          },
        },
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Fri Dec 21 12:50:07 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "media": Array [
              Object {
                "display_url": "pic.twitter.com/oWPdEZYeYf",
                "expanded_url": "https://twitter.com/todonoticias/status/1076097484445892608/photo/1",
                "id": 1076097429026558000,
                "id_str": "1076097429026557952",
                "indices": Array [
                  104,
                  127,
                ],
                "media_url": "http://pbs.twimg.com/media/Du8Q6QjW0AAqPX2.jpg",
                "media_url_https": "https://pbs.twimg.com/media/Du8Q6QjW0AAqPX2.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 765,
                    "resize": "fit",
                    "w": 1366,
                  },
                  "medium": Object {
                    "h": 672,
                    "resize": "fit",
                    "w": 1200,
                  },
                  "small": Object {
                    "h": 381,
                    "resize": "fit",
                    "w": 680,
                  },
                  "thumb": Object {
                    "h": 150,
                    "resize": "crop",
                    "w": 150,
                  },
                },
                "type": "photo",
                "url": "https://t.co/oWPdEZYeYf",
              },
            ],
            "symbols": Array [],
            "urls": Array [],
            "user_mentions": Array [],
          },
          "extended_entities": Object {
            "media": Array [
              Object {
                "display_url": "pic.twitter.com/oWPdEZYeYf",
                "expanded_url": "https://twitter.com/todonoticias/status/1076097484445892608/photo/1",
                "id": 1076097429026558000,
                "id_str": "1076097429026557952",
                "indices": Array [
                  104,
                  127,
                ],
                "media_url": "http://pbs.twimg.com/media/Du8Q6QjW0AAqPX2.jpg",
                "media_url_https": "https://pbs.twimg.com/media/Du8Q6QjW0AAqPX2.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 765,
                    "resize": "fit",
                    "w": 1366,
                  },
                  "medium": Object {
                    "h": 672,
                    "resize": "fit",
                    "w": 1200,
                  },
                  "small": Object {
                    "h": 381,
                    "resize": "fit",
                    "w": 680,
                  },
                  "thumb": Object {
                    "h": 150,
                    "resize": "crop",
                    "w": 150,
                  },
                },
                "type": "photo",
                "url": "https://t.co/oWPdEZYeYf",
              },
            ],
          },
          "favorite_count": 2,
          "favorited": false,
          "geo": null,
          "id": 1076097484445892600,
          "id_str": "1076097484445892608",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 1,
          "retweeted": false,
          "source": "<a href=\\"http://twitter.com\\" rel=\\"nofollow\\">Twitter Web Client</a>",
          "text": "Intentó violar a su maestra en plena cárcel de Mendoza: \\"Quedate bien quietita\\" https://t.co/4IElOV3zZ3 https://t.co/oWPdEZYeYf",
          "truncated": false,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Mon Mar 23 12:35:07 +0000 2009",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Noticias las 24 horas, por el canal líder en audiencia y credibilidad de Argentina",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "tn.com.ar",
                    "expanded_url": "http://www.tn.com.ar",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/TX6TF1nOUs",
                  },
                ],
              },
            },
            "favourites_count": 1,
            "follow_request_sent": false,
            "followers_count": 4678381,
            "following": true,
            "friends_count": 124,
            "geo_enabled": false,
            "has_extended_profile": false,
            "id": 25992212,
            "id_str": "25992212",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 9081,
            "location": "Argentina",
            "name": "TN - Todo Noticias",
            "notifications": false,
            "profile_background_color": "666666",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/25992212/1533675118",
            "profile_image_url": "http://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_link_color": "D51D2E",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "314BB9",
            "profile_text_color": "000000",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "todonoticias",
            "statuses_count": 84451,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/TX6TF1nOUs",
            "utc_offset": null,
            "verified": true,
          },
        },
      ]
    }
  />
</Component>
`;

exports[`post screen should render correctly when loading 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</Component>
`;
